### PR TITLE
Refactor nsexec.c and add some comments

### PR DESCRIPTION
**Pull request by hqhq**
Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

### CODE REVIEW

**Comment by hqhq** [comment]
Added a patch to close opened files before exit.

**Comment by mlaventure** [comment]
Looks good.

Could pass `struct nsenter_config` as a pointer to avoid a copy, but it's fairly small, so I guess it doesn't matter :)

**Comment by mikebrow** [comment]
Reviewed.. no issues found.  Looks good.

File: libcontainer/nsenter/nsexec.c [hunk 1]
```diff
@@ -344,14 +344,61 @@ static void process_nl_attributes(int pipenum, char *data, int data_size)
...
+
 	// required clone_flags to be passed
 	if (config.cloneflags == -1) {
 		pr_perror("Missing clone_flags");
 		exit(1);
 	}
 	// prepare sync pipe between parent and child. We need this to let the
-	// child
-	// know that the parent has finished setting up
+	// child know that the parent has finished setting up
```
Comment (mrunalp) [inline libcontainer/nsenter/nsexec.c]: s/know/knows

File: libcontainer/nsenter/nsexec.c [hunk 2]
```diff
@@ -344,14 +344,61 @@ static void process_nl_attributes(int pipenum, char *data, int data_size)
...
 
+	return config;
+}
+
+void nsexec(void)
+{
+	int pipenum;
+
+	// If we dont have init pipe, then just return to the go routine,
+	// we'll only have init pipe when do start or exec
```
Comment (mrunalp) [inline libcontainer/nsenter/nsexec.c]: s/when do/for/ ?

File: libcontainer/nsenter/nsexec.c (hunk 1 above)
Comment (mikebrow) [inline libcontainer/nsenter/nsexec.c]: "We need this to let the child know that the parent has finished setting up..."

it's correct as is :-)

Comment (mrunalp) [inline libcontainer/nsenter/nsexec.c]: @mikebrow yeah, my bad. I didn't read the previous line :)

**Comment by dqminh** [comment]
LGTM

File: libcontainer/nsenter/nsexec.c (hunk 2 above)
Comment (hqhq) [inline libcontainer/nsenter/nsexec.c]: Updated.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**